src/basic/pidref.c

```diff
@@ -428,6 +428,10 @@ int pidref_wait_for_terminate(const PidRef *pidref, siginfo_t *ret) {
         }
 }
 
+bool pidref_is_automatic(const PidRef *pidref) {
```

Comment:
I'd like to split the first commit into at least two, pidref and json part, then machined part.